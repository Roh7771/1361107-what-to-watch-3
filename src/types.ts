export interface Film {
  title: string;
  genre: string;
  releaseYear: number;
  imgSrc: string;
  bgSrc: string;
  posterSrc: string;
  ratingScore: number;
  ratingCount: number;
  description: string[];
  director: string;
  starring: string[];
  id: number;
  filmDuration: number;
  videoSrc: string;
  isFavorite: boolean;
  trailerSrc: string;
  bgColor: string;
}

export interface Comment {
  id: number;
  user: {
    id: number;
    name: string;
  };
  rating: number;
  comment: string;
  date: string;
}
