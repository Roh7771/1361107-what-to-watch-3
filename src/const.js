export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  MY_LIST: `/mylist`,
  FILM: `/films`,
  PLAYER: `/player`
};

export const MIN_TEXT_LENGTH = 50;

export const DEFAULT_STARS_NUMBER = 3;

export const MORE_LIKE_THIS_FILMS_COUNT = 4;

export const TRAILER_APPEARANCE_DELAY = 1000;

export const INIT_FILMS_TO_SHOW_NUMBER = 8;

export const FILMS_TO_ADD_NUMBER = 8;

