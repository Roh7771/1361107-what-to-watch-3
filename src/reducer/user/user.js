import {extend} from "../../utils";
import history from "../../history";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
};

const ActionTypes = {
  REQUIRE_AUTHORIZATION: `REQUIRE_AUTHORIZATION`,
};

const ActionCreators = {
  requireAuthorization: (status) => {
    return {
      type: ActionTypes.REQUIRE_AUTHORIZATION,
      payload: status,
    };
  },
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => {
        dispatch(ActionCreators.requireAuthorization(AuthorizationStatus.AUTH));
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then(() => {
        dispatch(ActionCreators.requireAuthorization(AuthorizationStatus.AUTH));
        history.push(`/`);
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.REQUIRE_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });
  }

  return state;
};

export {
  ActionCreators,
  ActionTypes,
  reducer,
  Operation,
  AuthorizationStatus
};
