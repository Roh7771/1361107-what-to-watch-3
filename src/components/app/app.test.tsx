import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {App} from "./app";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {AppRoute} from "../../const.js";
import {MemoryRouter} from "react-router-dom";
import {Film} from "../../types.js";
import {noop} from "../../utils";

const mockStore = configureStore([]);

const mock: {userFavoriteFilms: Film[]; promoFilm: Film; filmsList: Film[] } = {
  userFavoriteFilms: [
    {
      title: `Some Title`,
      trailerSrc: `some path`,
      genre: `Comedy`,
      bgColor: `red`,
      releaseYear: 2015,
      imgSrc: `Some Path`,
      bgSrc: `iSome Path`,
      posterSrc: `Some Path`,
      ratingScore: 8.7,
      ratingCount: 230,
      descriptionList: [
        `Some description`,
      ],
      director: `Some cool directot`,
      starring: [`Actor1`, `Actor2`],
      id: 2,
      videoSrc: `Some Path`,
      filmDuration: 99,
      isFavorite: false,
    }
  ],
  promoFilm: {
    title: `Some Title`,
    trailerSrc: `some path`,
    genre: `Comedy`,
    bgColor: `red`,
    releaseYear: 2015,
    imgSrc: `Some Path`,
    bgSrc: `iSome Path`,
    posterSrc: `Some Path`,
    ratingScore: 8.7,
    ratingCount: 230,
    descriptionList: [
      `Some description`,
    ],
    director: `Some cool directot`,
    starring: [`Actor1`, `Actor2`],
    id: 2,
    videoSrc: `Some Path`,
    filmDuration: 99,
    isFavorite: false,
  },
  filmsList: [
    {
      title: `Some Title`,
      trailerSrc: `some path`,
      genre: `Comedy`,
      bgColor: `red`,
      releaseYear: 2015,
      imgSrc: `Some Path`,
      bgSrc: `iSome Path`,
      posterSrc: `Some Path`,
      ratingScore: 8.7,
      ratingCount: 230,
      descriptionList: [
        `Some description`,
      ],
      director: `Some cool directot`,
      starring: [`Actor1`, `Actor2`],
      id: 2,
      videoSrc: `Some Path`,
      filmDuration: 99,
      isFavorite: false
    },
    {
      title: `Some Title`,
      trailerSrc: `some path`,
      genre: `Comedy`,
      releaseYear: 2015,
      imgSrc: `Some Path`,
      bgSrc: `iSome Path`,
      posterSrc: `Some Path`,
      ratingScore: 8.7,
      bgColor: `red`,
      ratingCount: 230,
      descriptionList: [
        `Some description`,
      ],
      director: `Some cool directot`,
      starring: [`Actor1`, `Actor2`],
      id: 4,
      videoSrc: `Some Path`,
      filmDuration: 99,
      isFavorite: false
    },
  ]
};

describe(`App should`, () => {
  it(`render main screen`, () => {
    const {filmsList, promoFilm, userFavoriteFilms} = mock;
    const store = mockStore({
      DATA: {
        filmsList,
        promoFilm,
      },
      APP_STATUS: {
        currentGenre: `All genres`,
        filmsToShowCount: 8,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter initialEntries={[AppRoute.ROOT]}>
              <App
                filmsToRender={filmsList}
                userFavoriteFilms={userFavoriteFilms}
                allFilms={filmsList}
                promoFilm={promoFilm}
                onLoginFormSubmit={noop}
                authorizationStatus={AuthorizationStatus.NO_AUTH}
                onFavoriteButtonClick={noop}
                onReviewSend={noop}
                isFormSending={false}
                isFilmsLoading={false}
                setFilmComments={noop}
                formErrorMessage={null}
                filmComments={[]}
              />
            </MemoryRouter>
          </Provider>,
          {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render auth screen`, () => {
    const {filmsList, promoFilm, userFavoriteFilms} = mock;
    const store = mockStore({
      DATA: {
        filmsList,
        promoFilm,
      },
      APP_STATUS: {
        currentGenre: `All genres`,
        filmsToShowCount: 8,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter initialEntries={[AppRoute.LOGIN]}>
              <App
                filmsToRender={filmsList}
                userFavoriteFilms={userFavoriteFilms}
                allFilms={filmsList}
                promoFilm={promoFilm}
                onLoginFormSubmit={noop}
                authorizationStatus={AuthorizationStatus.NO_AUTH}
                onFavoriteButtonClick={noop}
                onReviewSend={noop}
                isFormSending={false}
                isFilmsLoading={false}
                setFilmComments={noop}
                formErrorMessage={null}
                filmComments={[]}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render movie page screen`, () => {
    const {filmsList, promoFilm, userFavoriteFilms} = mock;
    const store = mockStore({
      DATA: {
        filmsList,
        promoFilm,
      },
      APP_STATUS: {
        currentGenre: `All genres`,
        filmsToShowCount: 8,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter initialEntries={[`${AppRoute.FILM}/2`]}>
              <App
                filmsToRender={filmsList}
                userFavoriteFilms={userFavoriteFilms}
                allFilms={filmsList}
                promoFilm={promoFilm}
                onLoginFormSubmit={noop}
                authorizationStatus={AuthorizationStatus.NO_AUTH}
                onFavoriteButtonClick={noop}
                onReviewSend={noop}
                isFormSending={false}
                isFilmsLoading={false}
                setFilmComments={noop}
                formErrorMessage={null}
                filmComments={[]}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render movie video player`, () => {
    const {filmsList, promoFilm, userFavoriteFilms} = mock;
    const store = mockStore({
      DATA: {
        filmsList,
        promoFilm,
      },
      APP_STATUS: {
        currentGenre: `All genres`,
        filmsToShowCount: 8,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter initialEntries={[`${AppRoute.PLAYER}/2`]}>
              <App
                filmsToRender={filmsList}
                userFavoriteFilms={userFavoriteFilms}
                allFilms={filmsList}
                promoFilm={promoFilm}
                onLoginFormSubmit={noop}
                authorizationStatus={AuthorizationStatus.NO_AUTH}
                onFavoriteButtonClick={noop}
                onReviewSend={noop}
                isFormSending={false}
                isFilmsLoading={false}
                setFilmComments={noop}
                formErrorMessage={null}
                filmComments={[]}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render user favorite films`, () => {
    const {filmsList, promoFilm, userFavoriteFilms} = mock;
    const store = mockStore({
      DATA: {
        filmsList,
        promoFilm,
      },
      APP_STATUS: {
        currentGenre: `All genres`,
        filmsToShowCount: 8,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter initialEntries={[`${AppRoute.MY_LIST}`]}>
              <App
                filmsToRender={filmsList}
                userFavoriteFilms={userFavoriteFilms}
                allFilms={filmsList}
                promoFilm={promoFilm}
                onLoginFormSubmit={noop}
                authorizationStatus={AuthorizationStatus.AUTH}
                onFavoriteButtonClick={noop}
                onReviewSend={noop}
                isFormSending={false}
                isFilmsLoading={false}
                setFilmComments={noop}
                formErrorMessage={null}
                filmComments={[]}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render review form`, () => {
    const {filmsList, promoFilm, userFavoriteFilms} = mock;
    const store = mockStore({
      DATA: {
        filmsList,
        promoFilm,
      },
      APP_STATUS: {
        currentGenre: `All genres`,
        filmsToShowCount: 8,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter initialEntries={[`${AppRoute.FILM}/2/review`]}>
              <App
                filmsToRender={filmsList}
                userFavoriteFilms={userFavoriteFilms}
                allFilms={filmsList}
                promoFilm={promoFilm}
                onLoginFormSubmit={noop}
                authorizationStatus={AuthorizationStatus.AUTH}
                onFavoriteButtonClick={noop}
                onReviewSend={noop}
                isFormSending={false}
                isFilmsLoading={false}
                setFilmComments={noop}
                formErrorMessage={null}
                filmComments={[]}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
