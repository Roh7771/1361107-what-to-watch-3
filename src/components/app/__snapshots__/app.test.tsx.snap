// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render auth screen 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <fieldset
        disabled={false}
        style={
          Object {
            "border": "none",
            "padding": 0,
          }
        }
      >
        <div
          className="sign-in__fields"
        >
          <div
            className="sign-in__field"
          >
            <input
              className="sign-in__input"
              id="user-email"
              name="user-email"
              placeholder="Email address"
              type="email"
            />
            <label
              className="sign-in__label visually-hidden"
              htmlFor="user-email"
            >
              Email address
            </label>
          </div>
          <div
            className="sign-in__field"
          >
            <input
              className="sign-in__input"
              id="user-password"
              name="user-password"
              placeholder="Password"
              type="password"
            />
            <label
              className="sign-in__label visually-hidden"
              htmlFor="user-password"
            >
              Password
            </label>
          </div>
        </div>
        <div
          className="sign-in__submit"
        >
          <button
            className="sign-in__btn"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </fieldset>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`App should render main screen 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Some Title"
        src="iSome Path"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Some Title poster"
            height="327"
            src="Some Path"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Some Title
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedy
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                className=""
                height={175}
                poster="Some Path"
                src="some path"
                width={280}
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Some Title
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                className=""
                height={175}
                poster="Some Path"
                src="some path"
                width={280}
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Some Title
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`App should render movie page screen 1`] = `
<div
  className="page-content"
>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Some Title"
          src="iSome Path"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Some Title
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedy
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/films/2/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Some Title poster"
            height="327"
            src="Some Path"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8,7
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                230 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              Some description
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: Some cool directot
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: Actor1, Actor2 and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <p>
          Sorry, there is no more films :(
        </p>
      </div>
    </section>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`App should render movie video player 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    height={null}
    poster="Some Path"
    src="Some Path"
    width={null}
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value="0"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        0:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Some Title
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`App should render review form 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Some Title"
        src="iSome Path"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/2"
              onClick={[Function]}
            >
              Some Title
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="/img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Some Title"
        height="327"
        src="Some Path"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            checked={false}
            className="rating__input"
            disabled={false}
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="1"
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 1
          </label>
          <input
            checked={false}
            className="rating__input"
            disabled={false}
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="2"
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 2
          </label>
          <input
            checked={true}
            className="rating__input"
            disabled={false}
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="3"
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 3
          </label>
          <input
            checked={false}
            className="rating__input"
            disabled={false}
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="4"
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 4
          </label>
          <input
            checked={false}
            className="rating__input"
            disabled={false}
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="5"
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          disabled={false}
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Text must contain at least 50 symbols"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;

exports[`App should render user favorite films 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/2"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className=""
              height={175}
              poster="Some Path"
              src="some path"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            Some Title
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
